pkg/arvo/app/chat-push-hook.hoon

```diff
@@ -0,0 +1,255 @@
+/-  *permission-store, *invite-store, hook=chat-push-hook
+/+  default-agent, verb, dbug, store=chat-store, lib=chat-hooks
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  [%0 state-0]
+  ==
++$  state-0
+  $:  sharing=(set path)
+      allow-history=(map path ?)
+  ==
+--
+::
+=|  [%0 state-0]
+=*  state  -
+::
+%-  agent:dbug
+%+  verb  |
+::
+^-  agent:gall
+|_  bol=bowl:gall
++*  this  .
+    def  ~(. (default-agent this %|) bol)
+::
+++  on-init
+  ^-  (quip card _this)
+  :_  this
+  [%pass /permissions %agent [our.bol %permission-store] %watch /updates]~
+::
+++  on-save  !>(state)
+++  on-load
+  |=  vax=vase
+  =/  old  !<(versioned-state vax)
+  ?-  -.old
+    %0  [~ this(state old)]
+  ==
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  =^  cards  state
+    ?+  mark  (on-poke:def mark vase)
+        %json  (chat-action (action:dejs:store !<(json vase)))
+        %chat-action  (chat-action !<(action:store vase))
+        %chat-push-hook-action  (chat-push-hook-action !<(action:hook vase))
+    ==
+  [cards this]
+  ::
+  ++  chat-action
+    |=  act=action:store
+    ^-  (quip card _state)
+    ::  XX why do we use ?> here but ?. ~ below?
```

Comment:
The thought is that we want it to loudly squawk at us if it gets unexpected input, but quietly decline if someone sends a message they’re not permitted to send.

Reply:
Don't you want it to be loud in both cases? This lets the sender/client know that it failed, instead of just wondering why they aren't getting their messages echoed back to them despite getting an ack on the poke. Not sure what you gain by hiding that.